src/shared/views/plugins/DeepZoom/DeepZoom.js

```diff
@@ -1,17 +1,43 @@
-import React from 'react'
+import React, { Component, PropTypes } from 'react'
+import OpenSeaDragon from 'openseadragon'
 
 import styles from './DeepZoom.css'
 
-const DeepZoom = (props) => {
-  return (
-    <div className={styles.holder}>
-      <iframe width="100%" height="100%" src={props.url} frameBorder="0" allowFullScreen />
-    </div>
-  )
-}
+const osdId = 'osd'
+
+class DeepZoom extends Component {
+  static propTypes = {
+    url: PropTypes.string.isRequired,
+  }
+
+  constructor(props) {
+    super(props)
+    this.state = {}
+  }
+
+  componentDidMount() {
+    this.onMount(imgUrl => {
+      this.setState({
+        viewer: OpenSeaDragon({
+          element: this[osdId],
+          prefixUrl: '/static/images/dzi-controls/',
+          tileSources: imgUrl,
+          maxZoomPixelRatio: Infinity,
+        }),
+      })
+    })
+  }
+
+  onMount(cb) { cb(this.props.url) }
 
-DeepZoom.propTypes = {
-  url: React.PropTypes.string.isRequired,
+  render() {
+    return (
+      <div
+        ref={elm => { this[osdId] = elm }}
+        className={styles.holder}
+      />
+    )
+  }
 }
 
 export default DeepZoom
```

Comment:
I would simplify this component.

 Maybe I am missing something and there is a part I dont understand?

Wanna chat on slack about it @gerardmrk ?

```js
import React from 'react'
import OpenSeaDragon from 'openseadragon'

import styles from './DeepZoom.css'

const DeepZoom = (props) => {
  const mountOpenSeaDragon = (el, url) => {
    OpenSeaDragon({
      element: el,
      prefixUrl: '/static/images/dzi-controls/',
      tileSources: url,
      maxZoomPixelRatio: Infinity,
    })
  }

  return (
    <div
      ref={el => { mountOpenSeaDragon(el, props.url) }}
      className={styles.holder}
    />
  )
}

DeepZoom.propTypes = {
  url: React.PropTypes.string.isRequired,
}

export default DeepZoom
```